test/src/edu/stanford/nlp/simple/SentenceTest.java

```diff
@@ -157,4 +167,120 @@ public void testTokenizeWhitespaceWithSpaces() {
     assertEquals("baz", s.word(2));
   }
 
+  @Test
+  public void testStringRepresentation() {
+    String initialSentence = "This is a sentence.";
+    Sentence sut = new Sentence(initialSentence);
+    String converted = sut.toString();
+    assertEquals(initialSentence, converted);
+  }
+
+  @Test
+  public void testSubstring() {
+    String initialSentence = "This is a sentence.";
+    Sentence sut = new Sentence(initialSentence);
+    String substring = sut.substring(1, 4);
+    assertEquals("is a sentence", substring);
+  }
+
+  @Test
+  public void testSentenceFromDocument() {
+    String initialDocument = "This is a sentence. "
+                           + "This is another sentence. "
+                           + "Together, they form a document!";
+    Document d = new Document(initialDocument);
+    int sentenceIndex = 2;
+    Sentence sut = new Sentence(d, sentenceIndex);
+    String sentence = sut.toString();
+    assertEquals("Together, they form a document!", sentence);
+  }
+
+  @Test
+  public void testSerializeToStream() throws IOException {
+    final class MockOutputStream extends OutputStream {
+      int writeCount = 0;
+        boolean closed = false;
+		
+        protected int getWriteCount () {
+          return writeCount;
+        }
+		
+        protected boolean gotClosed() {
+          return closed;
+        }
+        @Override
+        public void write(int arg0) throws IOException {
+          writeCount++;
+        }
+        @Override
+        public void close () throws IOException {
+          this.closed = true;	
+          super.close();
+        }
+    }
+    Sentence sut = new Sentence("This is a sentence.");
+    MockOutputStream mockedOutStream = new MockOutputStream();
+    sut.serialize(mockedOutStream);
+
+    /* The serialize method should write but should not close the
+     * provided OutputStream as specified in Sentence.java. */
+    boolean wasWrittenTo = mockedOutStream.getWriteCount() > 0;
+    boolean wasNotClosed = !mockedOutStream.gotClosed();
+
+    assertTrue(wasWrittenTo);
+    assertTrue(wasNotClosed);
+  }
+
+  @Test
+  public void testDeserializeFromInputStream () throws IOException {
```

Comment:
I really thought I just commented on this, but the comment is gone now - this seems difficult to change if the serialization format changes in the future.  I think it would work better if you first serialized a sentence, deserialized it, and compared the results.

Aside from that, I merged the changes for SentenceTest.  Thanks for the contribution!

Reply:
One of the other comments I had made was that a ByteArrayOutputStream or some similar built in class should be sufficient for this.